src/sqlancer/postgres/oracle/PostgresCERTOracle.java

```diff
@@ -0,0 +1,266 @@
+        if (increase) {
+            select.clearGroupByExpressions();
+        } else {
+            select.setGroupByExpressions(select.getFetchColumns());
+        }
+        return increase;
+    }
+
+    @Override
+    protected boolean mutateHaving() {
+        if (select.getGroupByExpressions().size() == 0) {
+            select.setGroupByExpressions(select.getFetchColumns());
+            select.setHavingClause(gen.generateExpression(0, PostgresDataType.BOOLEAN));
+            return false;
+        } else {
+            if (select.getHavingClause() == null) {
+                select.setHavingClause(gen.generateExpression(0, PostgresDataType.BOOLEAN));
+                return false;
+            } else {
+                select.setHavingClause(null);
+                return true;
+            }
+        }
+    }
+
+    @Override
+    protected boolean mutateAnd() {
+        if (select.getWhereClause() == null) {
+            select.setWhereClause(gen.generateExpression(0, PostgresDataType.BOOLEAN));
+        } else {
+            PostgresExpression newWhere = new PostgresBinaryLogicalOperation(select.getWhereClause(),
+                    gen.generateExpression(0, PostgresDataType.BOOLEAN), BinaryLogicalOperator.AND);
+            select.setWhereClause(newWhere);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean mutateOr() {
+        if (select.getWhereClause() == null) {
+            select.setWhereClause(gen.generateExpression(0, PostgresDataType.BOOLEAN));
+            return false;
+        } else {
+            PostgresExpression newWhere = new PostgresBinaryLogicalOperation(select.getWhereClause(),
+                    gen.generateExpression(0, PostgresDataType.BOOLEAN), BinaryLogicalOperator.OR);
+            select.setWhereClause(newWhere);
+            return true;
+        }
+    }
+
+    @Override
+    protected boolean mutateLimit() {
+        boolean increase = select.getLimitClause() != null;
+        if (increase) {
+            select.setLimitClause(null);
+        } else {
+            Randomly r = new Randomly();
+            select.setLimitClause(PostgresConstant.createIntConstant((int) Math.abs(r.getInteger())));
+        }
+        return increase;
+    }
+
+    private int getRow(SQLGlobalState<?, ?> globalState, String selectStr, List<String> queryPlanSequences)
+            throws AssertionError, SQLException {
+        int row = -1;
+        String explainQuery = "EXPLAIN " + selectStr;
+
+        if (globalState.getOptions().logEachSelect()) {
+            globalState.getLogger().writeCurrent(explainQuery);
+            try {
+                globalState.getLogger().getCurrentFileWriter().flush();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        // Get row count
+        SQLQueryAdapter q = new SQLQueryAdapter(explainQuery, errors);
+        try (SQLancerResultSet rs = q.executeAndGet(globalState)) {
```

Comment:
It would be great to subsequently factor out some of this functionality and write some tests so that debugging any breaking changes in the query plan format are easier to debug.

Reply:
Ok, I'll look into this.